tests/components/zha/test_cover.py

```diff
@@ -129,3 +158,151 @@ async def get_chan_attr(*args, **kwargs):
+    assert hass.states.get(entity_id).state == STATE_CLOSED
+
+    # test to see if it opens
+    await send_attributes_report(hass, cluster_on_off, {8: 0, 0: True, 1: 1})
+    assert hass.states.get(entity_id).state == STATE_OPEN
+
+    # close from UI command fails
+    with patch("zigpy.zcl.Cluster.request", side_effect=asyncio.TimeoutError):
+        await hass.services.async_call(
+            DOMAIN, SERVICE_CLOSE_COVER, {"entity_id": entity_id}, blocking=True
+        )
+        assert cluster_on_off.request.call_count == 1
+        assert cluster_on_off.request.call_args[0][0] is False
+        assert cluster_on_off.request.call_args[0][1] == 0x0000
+        assert hass.states.get(entity_id).state == STATE_OPEN
+
+    with patch(
+        "zigpy.zcl.Cluster.request", AsyncMock(return_value=[0x1, zcl_f.Status.SUCCESS])
+    ):
+        await hass.services.async_call(
+            DOMAIN, SERVICE_CLOSE_COVER, {"entity_id": entity_id}, blocking=True
+        )
+        assert cluster_on_off.request.call_count == 1
+        assert cluster_on_off.request.call_args[0][0] is False
+        assert cluster_on_off.request.call_args[0][1] == 0x0000
+        assert hass.states.get(entity_id).state == STATE_CLOSED
+
+    # open from UI command fails
+    assert ATTR_CURRENT_POSITION not in hass.states.get(entity_id).attributes
+    await send_attributes_report(hass, cluster_level, {0: 0})
+    with patch("zigpy.zcl.Cluster.request", side_effect=asyncio.TimeoutError):
+        await hass.services.async_call(
+            DOMAIN, SERVICE_OPEN_COVER, {"entity_id": entity_id}, blocking=True
+        )
+        assert cluster_on_off.request.call_count == 1
+        assert cluster_on_off.request.call_args[0][0] is False
+        assert cluster_on_off.request.call_args[0][1] == 0x0001
+        assert hass.states.get(entity_id).state == STATE_CLOSED
+
+    # open from UI succeeds
+    with patch(
+        "zigpy.zcl.Cluster.request", AsyncMock(return_value=[0x0, zcl_f.Status.SUCCESS])
+    ):
+        await hass.services.async_call(
+            DOMAIN, SERVICE_OPEN_COVER, {"entity_id": entity_id}, blocking=True
+        )
+        assert cluster_on_off.request.call_count == 1
+        assert cluster_on_off.request.call_args[0][0] is False
+        assert cluster_on_off.request.call_args[0][1] == 0x0001
+        assert hass.states.get(entity_id).state == STATE_OPEN
+
+    # set position UI command fails
+    with patch("zigpy.zcl.Cluster.request", side_effect=asyncio.TimeoutError):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_SET_COVER_POSITION,
+            {"entity_id": entity_id, "position": 47},
+            blocking=True,
+        )
+        assert cluster_level.request.call_count == 1
+        assert cluster_level.request.call_args[0][0] is False
+        assert cluster_level.request.call_args[0][1] == 0x0004
+        assert int(cluster_level.request.call_args[0][3] * 100 / 255) == 47
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5575421)